src/context.jl

```diff
@@ -471,8 +471,8 @@ Note that unlike `overdub`, `fallback`, etc., this function is not intended to b
 
 See also:  [`overdub`](@ref), [`fallback`](@ref), [`recurse`](@ref)
 """
-@inline canrecurse(ctx::Context, f, ::Vararg{Any}) = !(isa(untag(f, ctx), Core.Builtin) || _iscompilerfunc(untag(f, ctx)))
-@inline canrecurse(ctx::Context, ::typeof(Core._apply), f, args...) = Core._apply(canrecurse, (ctx, f), args...)
-@inline canrecurse(ctx::Context, ::typeof(Core.invoke), f, _, args...) = canrecurse(ctx, f, args...)
+@inline canrecurse(ctx::Context, f, @nospecialize(args...)) = !(isa(untag(f, ctx), Core.Builtin) || _iscompilerfunc(untag(f, ctx)))
```

Comment:
I thought that Julia didn't specialize on unused arguments? It's a straight-forward optimization. Did you see a performance difference from that change?

Reply:
I did the experiments a while ago. and apparently did not keep notes.
Beyond the "After" screenshot from the profiler in the top post.

Even if this one didn't help the other two do, so having it match is nice for symetry.

This whole change definately gave solid improvements